relentless/potential/pair.py

```diff
@@ -561,3 +561,230 @@ def _derivative(self, param, r, epsilon, kappa, **params):
         if s:
             d = d.item()
         return d
+
+class Depletion(PairPotential):
+    r"""Depletion pair potential.
+
+    .. math::
+
+        u(r) = -\frac{\pi P(\frac{1}{2}(\sigma_i+\sigma_j)+\sigma_d-r)^2(r^2+r(\sigma_i+\sigma_j+2\sigma_d)-\frac{3}{4}(\sigma_i-\sigma_j)^2}{12r}
+
+    The required coefficients per pair are:
+
+    - :math:`P`: depletant osmotic pressure
+    - :math:`\sigma_i`: particle type `i` diameter
+    - :math:`\sigma_j`: particle type `j` diameter
+    - :math:`\sigma_d`: depletant diameter
+
+    The optional coefficients per pair are:
+
+    - ``rmin``: minimum radius, energy and force are 0 for ``r < rmin``. Ignored if ``False`` (default).
+    - ``rmax``: maximum radius, energy and force are 0 for ``r > rmax`` Ignored if ``False`` (default).
+    - ``shift``: If ``True``, shift potential to zero at ``rmax`` (default is ``False``).
+
+    Parameters
+    ----------
+    types : array_like
+        List of types (A type must be a str).
+
+    """
+    def __init__(self, types, shift=False):
+        super().__init__(types=types, params=('P','sigma_i','sigma_j','sigma_d'))
+
+    def _energy(self, r, P, sigma_i, sigma_j, sigma_d, **params):
+        """Evaluates the depletion potential energy.
+
+        Parameters
+        ----------
+        r : array_like
+            The value or values of r at which to evaluate the energy.
+        P : float or int
+            The P parameter for the potential function.
+        sigma_i : float or int
+            The sigma_i parameter for the potential function.
+        sigma_j : float or int
+            The sigma_j parameter for the potential function.
+        sigma_d : float or int
+            The sigma_d parameter for the potential function.
+
+        Returns
+        -------
+        array_like
+            Returns the value or values of the energy evaluated at r.
+
+        Raises
+        ------
+        ValueError
+            If sigma_i, sigma_j, and sigma_d are not all positive.
+
+        """
+        if sigma_i<=0 or sigma_j<=0 or sigma_d<=0:
+            raise ValueError('sigma_i, sigma_j, and sigma_d must all be positive')
+        r,u,s = self._zeros(r)
+
+        #clamp lo
+        lo = 0.5*(sigma_i + sigma_j)
+        u_lo = -np.pi*P*sigma_d**3/6*(1 + 3*sigma_i*sigma_j/(sigma_d*(sigma_i + sigma_j)))
+        f_lo  = self._force(r=lo, P=P, sigma_i=sigma_i, sigma_j=sigma_j, sigma_d=sigma_d)
+        u[np.less(r,lo)] = u_lo - f_lo*(r[np.less(r,lo)] - lo)
+
+        #clamp hi
+        hi = lo + sigma_d
+        u[np.greater(r,hi)] = 0
+
+        # evaluate in between
+        flags = np.logical_and(np.greater_equal(r,lo), np.less_equal(r,hi))
```

Comment:
This will give you exactly all of the points that were not below / above (first OR those flags together, then negate the result).
```suggestion
        flags = ~np.logical_or(below,above)
```